ecs-agent/tmds/handlers/fault/v1/types/types.go

```diff
@@ -102,29 +102,39 @@ func (request NetworkLatencyRequest) ValidateRequest() error {
 	if request.Sources == nil || len(request.Sources) == 0 {
 		return fmt.Errorf(missingRequiredFieldError, "Sources")
 	}
-	for _, element := range request.Sources {
-		elementStr := aws.StringValue(element)
-		validIp := true
-		if net.ParseIP(elementStr) == nil {
-			validIp = false
-		}
-		validIpCIDRBlock := true
-		if _, _, err := net.ParseCIDR(elementStr); err != nil {
-			validIpCIDRBlock = false
-		}
+	return validateNetworkFaultRequestSources(request.Sources)
+}
 
-		if !validIpCIDRBlock && !validIp {
-			return fmt.Errorf(invalidValueError, elementStr, "Sources")
-		}
+func (request NetworkLatencyRequest) ToString() string {
+	data, err := json.Marshal(request)
+	if err != nil {
+		return fmt.Sprintf("Error: Unable to parse %s request with error %v.", LatencyFaultType, err)
 	}
+	return string(data)
+}
 
-	return nil
+// NetworkPacketLossRequest is struct for the network packet loss fault request.
+type NetworkPacketLossRequest struct {
+	LossPercent *uint64 `json:"LossPercent"`
+	// Sources is a list including IPv4 addresses or IPv4 CIDR blocks.
+	Sources []*string `json:"Sources"`
 }
 
-func (request NetworkLatencyRequest) ToString() string {
+// ValidateRequest validates required fields are present and its value.
+func (request NetworkPacketLossRequest) ValidateRequest() error {
+	if request.LossPercent == nil {
```

Comment:
Can we check its value as well? this should be percentage - https://docs.aws.amazon.com/fis/latest/userguide/fis-actions-reference.html#task-network-packet-loss

Reply:
Thanks for the call out. I think we can assert the following: 0 <= request.LossPercent <= 100, which validates that this is a percentage (without the percentage sign)

Reply:
it should be 1 for the minimum value

Reply:
ACK. Just updated the PR with new revision. Added check to ensure value is within [1, 100], and added 3 unhappy test cases: when the value is -1 (fails with unable to store -1 in uint64), when the value is 101, and when the value is 0 (both fails with invalid val)